### PR TITLE
DYN-4949 Develop test framework to test correctness of graph results

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

This pull request does:
* add a test framework for the new engine work.
* use WorkspaceComparisonData for comparing the results from two runs of the same graph.
* do the comparison in memory.
* restart Dynamo between each run,
* not add a way to switch between the old and new engine. This is future work.
* target the emitMSIL branch. Testing was done using the master branch.

This eventually boiled down to be very little code. I opted for comparing the run results in memory. This assumes that we will have a way to switch between the engines at runtime. We can always change this and start serialize the results if we need to compare the results from different binaries.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/DynamoCoreTests/Performance/PerformanceTests.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        [TestFixtureTearDown]
+        public void TearDown()
+        {
+            ExecutionEvents.GraphPostExecution -= ExecutionEvents_GraphPostExecution;
+        }
+
+        private void ExecutionEvents_GraphPostExecution(Session.IExecutionSession session)
+        {
+            lastExecutionDuration = (TimeSpan)session.GetParameterValue(Session.ParameterKeys.LastExecutionDuration);
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/Performance/PerformanceTests.cs:48]: @sm6srw did you intend to use this `lastExecutionDuration` timespan somewhere?

Comment (sm6srw) [reply test/DynamoCoreTests/Performance/PerformanceTests.cs:48]: Yes, but I forgot about it.

File: test/DynamoCoreTests/Performance/PerformanceTests.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        private void DoWorkspaceOpenAndCompare(string filePath)
+        {
+            OpenModel(filePath);
+
+            var model = CurrentDynamoModel;
+            var ws1 = model.CurrentWorkspace;
+            ws1.Description = "TestDescription";
+
+            var dummyNodes = ws1.Nodes.Where(n => n is DummyNode);
+            if (dummyNodes.Any())
```
Comment (pinzart90) [inline test/DynamoCoreTests/Performance/PerformanceTests.cs:66]: you can probably move some of this code to a reusable function (code is repeated after line 97)

**Review by pinzart90 — APPROVED** [review]

File: test/DynamoCoreTests/Performance/PerformanceTests.cs (hunk 1 above)
Comment (sm6srw) [reply test/DynamoCoreTests/Performance/PerformanceTests.cs]: Removed the timer for now.

File: test/DynamoCoreTests/Performance/PerformanceTests.cs (hunk 2 above)
Comment (sm6srw) [reply test/DynamoCoreTests/Performance/PerformanceTests.cs]: Yep

**Pull request merged**